torch/csrc/jit/fuser/cpu/temp_file.h

```diff
@@ -44,17 +149,35 @@ struct TempFile {
     AT_ASSERT(str.size() == result);
   }
 
+#ifdef _MSC_VER
+  void close() {
+    if (file_ != nullptr) {
+      fclose(file_);
+    }
+    file_ = nullptr;
+  }
+#endif
+
   FILE* file() {
     return file_;
   }
 
   ~TempFile() {
+#ifdef _MSC_VER
+    if (!name_.empty() && _access(name_.c_str(), 0) != -1) {
+      _unlink(name_.c_str());
```

Comment:
Err, aren't you not allowed to unlink the file until you've closed it? Am I missing something here?

Reply:
Normally, the file is closed before `_unlink` is called because I call `close` explicitly. But yes, we should swap the sequence here if only the destructor is called.

Reply:
Yeah, it seems safer to swap. It wasn't too clear to me why you are closing explicitly but I guess that's fine.

Reply:
Because on Windows, a file cannot be opened at the same time. So we must close it before compiling.

Reply:
...whooooaaaaaaa. I knew that you can't delete a file that's open, but you can't open a file at the same time by two programs? TIL! (Can you open a file by two programs as read only?)

Reply:
Yes, this can be done. But actually I cannot control how MSVC opens that file.